core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -487,7 +482,7 @@ public List<NakadiRecordResult> sendEvents(final String topic,
         } catch (final InterruptException | InterruptedException e) {
             Thread.currentThread().interrupt();
             return prepareResponse(nakadiRecords, responses, e);
-        } catch (final RuntimeException e) {
+        } catch (final RuntimeException | IOException e) {
             kafkaFactory.terminateProducer(producer);
```

Comment:
it looks like IOException is thrown whenever NakadiRecord was failed to be serialised to bytes. In that case there is no need to terminate the producer.

Reply:
This is fixed as well.